fusedrug_examples/interaction/drug_target/affinity_prediction/t5_mol/tokenizer/multi_tokenizer.py

```diff
@@ -0,0 +1,1304 @@
+
+        Returns:
+            bool: _description_
+        """
+        raise Exception("Not implemented")
+
+    @staticmethod
+    def cfg_list_2_dict(dict_list: List) -> Dict[str, Any]:
+        """Receives a list of dicts, each containing a key "name" and changes it to
+
+        Args:
+            dict_list (List): _description_
+
+        Returns:
+            Dict[str, Any]: _description_
+        """
+        return {d["name"]: d for d in dict_list}
+
+    def save_jsons(self, tokenizers_info: Optional[List] = None) -> None:
+        """_summary_
+
+        Args:
+            tokenizers_info_list (Optional[List], optional): A list of dictionaries containing the following:
+            [
+                   {
+                        "name": a name of a tokenizer
+                        "modular_json_path":out_path for tokenizer_type
+                    }
+            ]
+            In case of None, paths stored in self.tokenizers_info (modular_json_path for each tokenizer) are used.
+            In case of partial tokenizer_types, only those tokenizers will be saved
+            Defaults to None.
+            TODO: also save the config yaml there
+        """
+        if tokenizers_info is None:
+            for t_type in self.tokenizers_info:
+                tokenizer_inst = self.tokenizers_info[t_type]["tokenizer_inst"]
+                out_path = self.tokenizers_info[t_type]["modular_json_path"]
+                tokenizer_inst.save(out_path)
+        else:
+            tokenizers_info_dict = ModularTokenizer.cfg_list_2_dict(tokenizers_info)
+            for t_type in tokenizers_info_dict:
+                tokenizer_inst = self.tokenizers_info[t_type]["tokenizer_inst"]
+                out_path = tokenizers_info_dict[t_type]["modular_json_path"]
+                tokenizer_inst.save(out_path)
+
+    def _add_single_tokenizer(
+        self,
+        tokenizer_info: Dict,
+    ) -> None:
+        raise Exception("Not implemented")
+
+    def add_tokenizers(
+        self,
+    ) -> None:
+        raise Exception("Not implemented")
+
+    def _encode_single_type(self, data_str: str, input_type: str, sequence_id: Optional[int] = None) -> Encoding:
+        assert isinstance(data_str, str)
+        assert isinstance(input_type, str)
+
+        if input_type not in self.tokenizers_info:
+            raise Exception(f"Input type {input_type} not found")
+
+        encoded = self.tokenizers_info[input_type]["tokenizer_inst"].encode(data_str)
+
+        if len(encoded.overflowing) > 0:
+            print(
+                f"Warning: FastTokenizer had to truncate sequence. Original Sequence Length = {len(data_str)}, max tokens supported = {self.tokenizers_info[input_type]['max_len']}, exceeded by {len(encoded.overflowing[0].ids)} tokens, for tokenizer: {input_type}"
+            )
+
+        if sequence_id is None:
+            sequence_id = int(self.tokenizers_info[input_type]["tokenizer_id"])  # this does not work.
```

Comment:
What do you plan to do if it doesn't work? :)

Reply:
Updated the comment. It may be a problem with set_sequence_id having a bug. The only way this can be used is if the sequence ids are a sequence of integers starting with 0. There isn't much to do, since we can't set sequence_id of a tokenizer instance directly, so we should keep this in mind should we ever use this functionality.